sycl/source/detail/graph_impl.cpp

```diff
@@ -1000,6 +1000,7 @@ exec_graph_impl::enqueue(const std::shared_ptr<sycl::detail::queue_impl> &Queue,
       }
 
       NewEvent = CreateNewEvent();
+      assert(NewEvent);
```

Comment:
I'd leave this assert out, I don't think it really adds any value

```cpp
  auto CreateNewEvent([&]() {
    auto NewEvent = std::make_shared<sycl::detail::event_impl>(Queue);
    NewEvent->setContextImpl(Queue->getContextImplPtr());
    NewEvent->setStateIncomplete();
    return NewEvent;
  });
```
We're already dereferencing the pointer inside `CreateNewEvent` before we reach this line of code